sdk/core/azure-core-amqp/inc/azure/core/amqp/message_receiver.hpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-Licence-Identifier: MIT
+
+#pragma once
+
+#include "claims_based_security.hpp"
+#include "common/async_operation_queue.hpp"
+#include "connection_string_credential.hpp"
+#include "link.hpp"
+#include "models/amqp_message.hpp"
+#include "models/amqp_value.hpp"
+#include "session.hpp"
+#include <azure/core/credentials/credentials.hpp>
+#include <azure/core/nullable.hpp>
+#include <vector>
+
+namespace Azure { namespace Core { namespace Amqp {
+  namespace _detail {
+    class MessageReceiverImpl;
+  }
+  namespace _internal {
+    enum class MessageReceiverState
+    {
+      Invalid,
+      Idle,
+      Opening,
+      Open,
+      Closing,
+      Error
+    };
+
+    enum class ReceiverSettleMode
+    {
+      First,
+      Second
+    };
+
+    class MessageReceiver;
+
+    struct MessageReceiverOptions
+    {
+      std::string Name;
+      std::vector<std::string> AuthenticationScopes;
+      ReceiverSettleMode SettleMode{ReceiverSettleMode::First};
+      std::string TargetAddress;
+      bool EnableTrace{false};
+      Azure::Nullable<uint32_t> InitialDeliveryCount;
+      Azure::Nullable<uint64_t> MaxMessageSize;
+
+      bool Batching{false};
+      std::chrono::seconds BatchMaxAge{std::chrono::seconds(5)};
+      std::vector<std::string> Capabilities;
+      uint32_t Credit{1};
+      LinkDurability Durability{};
+      bool DynamicAddress{false};
+      ExpiryPolicy SenderExpiryPolicy{};
+      ExpiryPolicy ReceiverExpiryPolicy{};
+      std::chrono::seconds ExpiryTimeout{std::chrono::seconds(0)};
+      // LinkFilter
+      bool ManualCredits{};
```

Comment:
As someone who isn't well-versed on AMQP, it isn't clear to me what some of these options are meant for, and I wonder if many of these are even required for a particular scenario. Maybe documentation would help, or alternatively, we leave some out until a use case arises, rather than be exhaustively complete right out of the gate.

For example, what is Credits/ManualCredits and BatchMaxeAge?

Reply:
#4503